modules/bmo-about-the-hostfirmwaresettings-resource.adoc

```diff
@@ -0,0 +1,83 @@
+// This is included in the following assemblies:
+//
+// post_installation_configuration/bare-metal-configuration.adoc
+
+:_content-type: REFERENCE
+[id="about-the-hostfirmwaresettings-resource_{context}"]
+= About the HostFirmwareSettings resource
+
+You can use the `HostFirmwareSettings` resource to retrieve and manage the BIOS settings for a host. When a host moves to the `Available` state, Ironic reads the host's BIOS settings and creates the `HostFirmwareSettings` resource. The resource contains the complete BIOS configuration returned from the baseboard management controller (BMC). Whereas, the `firmware` field in the `BareMetalHost` resource returns three vendor-independent fields, the `HostFirmwareSettings` resource typically comprises many BIOS settings of vendor-specific fields per host.
+
+The `HostFirmwareSettings` resource contains two sections:
+
+. The host firmware specification.
+. The host firmware's current status.
+
+[discrete]
+== The `HostFirmwareSettings` specification
+
+The `spec` section of the `HostFirmwareSettings` resource defines the desired state of the host's BIOS, and is empty by default. Ironic uses the settings in the `spec.settings` section to update the baseboard management controller (BMC) when the host is in the `Preparing` state. The `spec.settings` section uses the `FirmwareSchema` resource schema to ensure that you do not send invalid name/value pairs to hosts. See "About the FirmwareSchema resource" for additional details.
+
+.Example
+[source,terminal]
+----
+spec:
+  settings:
+    ProcTurboMode: Disabled<1>
+----
+<1> In the foregoing example, the `spec.settings` section contains a name/value pair that will set the `ProcTurboMode` BIOS setting to `Disabled`.
+
+[NOTE]
+====
+Integer parameters listed in the `status` section appear as strings. For example, `"1"`. When setting integers in the `spec.settings` section, the values should be set as integers without quotes. For example, `1`.
+====
+
+[discrete]
+== The `HostFirmwareSettings` status
+
+The `status` represents the current state of the host's BIOS.
+
+.HostFirmwareSettings
+[options="header"]
+|====
+|Parameters|Description
+a|
+----
+status:
+  conditions:
+  - lastTransitionTime:
+    message:
+    observedGeneration:
+    reason:
+    status:
+    type:
+----
+a| The `conditions` field contains a list of state changes. The sub-fields include:
+
+* `lastTransitionTime`: The last time the state changed.
+* `message`: A description of the state change.
+* `observedGeneration`: The current generation of the `status`. If `metadata.generation` and this field are not the same, the `status.conditions` might be out of date.
+* `reason`: The reason for the state change.
+* `status`: The status of the state change. The status can be `True`, `False` or `Unknown`.
+* `type`: The type of state change.
+
+a|
+----
+status:
+  schema:
+    name:
+    namespace:
+----
```

Comment:
There is an additional status file to note - lastUpdated, which is the last time the this resource was updated, e.g.
lastUpdated: "2022-02-17T02:38:40Z"

Reply:
Done